pyvista/plotting/lights.py

```diff
@@ -0,0 +1,465 @@
+        position and focal point can be defined in terms of an elevation
+        and an azimuthal angle, both in degrees.
+
+        Parameters
+        ----------
+        elev : float
+            The elevation of the directional light.
+
+        azim : float
+            The azimuthal angle of the directional light.
+
+        """
+        self.SetDirectionAngle(elev, azim)
+
+    # TODO: deepcopy?
+
+    def set_headlight(self):
+        """Set the light to be a headlight.
+
+        Headlights are fixed to the camera and always point to the focal
+        point of the camera. Calling this method will reset the light's
+        transformation matrix.
+
+        """
+        self.SetLightTypeToHeadlight()
+
+    def set_camera_light(self):
+        """Set the light to be a camera light.
+
+        Camera lights are fixed to the camera and always point to the focal
+        point of the camera.
+
+        A camera light moves with the camera, but it can have an arbitrary
+        relative position to the camera. Camera lights are defined in a
+        coordinate space where the camera is located at (0, 0, 1), looking
+        towards (0, 0, 0) at a distance of 1, with up being (0, 1, 0).
+        Camera lights use the transform matrix to establish this space.
+        Calling this method will reset the light's transformation matrix.
+
+        """
+        self.SetLightTypeToCameraLight()
+
+    def set_scene_light(self):
+        """Set the light to be a scene light.
+
+        Scene lights are stationary with respect to the scene.
+        Calling this method will reset the light's transformation matrix.
+
+        """
+        self.SetLightTypeToSceneLight()
+
+    @classmethod
+    def from_vtk(cls, vtk_light):
+        """Create a Light object from a vtkLight, resulting in a copy.
+
+        Parameters
+        ----------
+        vtk_light : vtkLight
+            The ``vtkLight`` to be copied.
+
+        """
+        if not isinstance(vtk_light, vtkLight):
+            raise TypeError(f'Expected ``vtkLight`` object, got ``{type(vtk_light)}`` instead.')
+
+        light = cls()
+        light.light_type = vtk_light.GetLightType()  # resets transformation matrix!
+        light.position = vtk_light.GetPosition()
+        light.focal_point = vtk_light.GetFocalPoint()
+        light.ambient_color = vtk_light.GetAmbientColor()
+        light.diffuse_color = vtk_light.GetDiffuseColor()
+        light.specular_color = vtk_light.GetSpecularColor()
+        light.intensity = vtk_light.GetIntensity()
+        light.is_on = vtk_light.GetSwitch()
+        light.positional = vtk_light.GetPositional()
+        light.exponent = vtk_light.GetExponent()
+        light.cone_angle = vtk_light.GetConeAngle()
+        light.attenuation_values = vtk_light.GetAttenuationValues()
+        light.shadow_attenuation = vtk_light.GetShadowAttenuation()
+        # TODO: copy transformation matrix even if not exposed?
+
+        return light
```

Comment:
I read from bane(I think) that pyvista objects should just handle what is thrown at them from the constructor, instead of classmethods. Though I disagree, this isn't how the others are constructed currently.

Reply:
And I absolutely agree! But I opted not to do that exactly because I saw that the rest of the API is not designed that way :) My preference would be to have a single optional positional-only `vtkLight` argument, and everything else keyword-only arguments.